DetourCrowd/Source/DetourProximityGrid.cpp

```diff
@@ -163,9 +163,9 @@ int dtProximityGrid::queryItems(const float minx, const float miny,
 					// Item not found, add it.
 					if (i == end)
 					{
+						ids[n++] = item.id;
```

Comment:
If `maxIds` is initially 0 and `ids` has 0 size (i.e. points to invalid memory) or `nullptr` this will lead to a crash due to `nullptr` dereference or out of bounds access.